### ISSUE TITLE
Should useWebKit be default?

### ISSUE THREAD

**Issue by scarlac** [react-native-cookies/cookies#69]
# Question

Should we default `useWebKit` to `true` as the default for all static methods? (e.g. `CookieManager.clearAll(true)`) Given that react-native-webview from the community has now removed UIWebView entirely ([link](https://github.com/react-native-community/react-native-webview#versioning)), it seems it would make sense to default to using WKWebView (aka `useWebKit`).

This would be a breaking chance, so despite 4.0 just being release it would probably to wait until 5.0 to be released.

Thoughts?

**Comment by davidsharp** [comment]
I think it's the sensible default going forward, also worth noting is the caveat that `setFromResponse` and `getFromResponse` are missing this functionality altogether (#72)

**Comment by davidgovea** [comment]
I had the similar idea, but after looking into it, the "default" mode (`NSHTTPCookieStorage`) is used for react-native's network stack (i.e. `fetch()`, etc)

That would make changing the default quite a large change for anyone relying on that behavior (as opposed to just webview cookie management). Not sure if that is a common use-case, but I suppose the change would be OK if it is rare.

I actually need to manage the cookies on "both sides" in iOS, so I've been using hacks like:

```js
for (const useWebkit of [false, true]) {
  // Call both non-webkit and webkit variants on iOS, but only once on Android
  if (!useWebkit || Platform.OS === 'ios') {
    CookieManager.set(url, cookie, useWebkit);
  }
}
```

To me, it would be nice to auto-sync the cookies between the two stores on iOS -- but again, not sure if my usecase is common.

**Comment by safaiyeh** [comment]
Yeah the reason behind not defaulting to WebKit is that fetch is using `NSHTTPCookieStorage`. 

There might be a cleaner design for this. Maybe even moving WebView related cookies to the WebView module https://github.com/react-native-community/cookies/issues/5

Looking to hear from everyone!

**Comment by scarlac** [comment]
Ah that makes sense now. I think just documenting it clearly would suffice. It's not 100% clear how they are shared so i had to look into Apple docs and read source to confirm. Not the fault of this lib but I can see others being confused by it.

**Comment by safaiyeh** [comment]
Sweet ill take this up as a documentation task

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to:
- [`@preeternal/react-native-cookie-manager`](https://github.com/Preeternal/react-native-cookie-manager)
- [`react-native-nitro-cookies`](https://github.com/l2hyunwoo/react-native-nitro-cookies)

Thank you for your contributions! ❤️
